irispy/new_sji.py

```diff
@@ -0,0 +1,177 @@
+'''
+This module provides movie tools for level 2 IRIS SJI fits file
+'''
+
+from datetime import timedelta
+
+from ndcube import NDCube
+import numpy as np
+from astropy.io import fits
+import astropy.units as u
+from astropy.wcs import WCS
+from sunpy.time import parse_time
+from sunpy import config
+from irispy import iris_tools
+
+__all__ = ['SJICube']
+
+TIME_FORMAT = config.get("general", "time_format")
+
+# the following value is only appropriate for byte scaled images
+BAD_PIXEL_VALUE = -200
+# the following value is only appropriate for unscaled images
+BAD_PIXEL_VALUE_UNSCALED = -32768
+
+
+class SJICube(NDCube):
+    """
+    SJICube
+
+    Class representing SJI Images described by a single WCS
+
+    Parameters
+    ----------
+    data: `numpy.ndarray`
+        The array holding the actual data in this object.
+
+    wcs: `ndcube.wcs.wcs.WCS`
+        The WCS object containing the axes' information
+
+    unit : `astropy.unit.Unit` or `str`
+        Unit for the dataset.
+        Strings that can be converted to a Unit are allowed.
+
+    meta : dict-like object
+        Additional meta information about the dataset.
+
+    uncertainty : any type, optional
+        Uncertainty in the dataset. Should have an attribute uncertainty_type
+        that defines what kind of uncertainty is stored, for example "std"
+        for standard deviation or "var" for variance. A metaclass defining
+        such an interface is NDUncertainty - but isn’t mandatory. If the
+        uncertainty has no such attribute the uncertainty is stored as
+        UnknownUncertainty.
+        Defaults to None.
+
+    mask : any type, optional
+        Mask for the dataset. Masks should follow the numpy convention
+        that valid data points are marked by False and invalid ones with True.
+        Defaults to None.
+
+    extra_coords : iterable of `tuple`s, each with three entries
+        (`str`, `int`, `astropy.units.quantity` or array-like)
+        Gives the name, axis of data, and values of coordinates of a data axis
+        not included in the WCS object.
+
+    copy : `bool`, optional
+        Indicates whether to save the arguments as copy. True copies every
+        attribute before saving it while False tries to save every parameter
+        as reference. Note however that it is not always possible to save the
+        input as reference.
+        Default is False.
+
+    Examples
+    --------
+    >>> from irispy import sji
+    >>> from irispy.data import sample
+    >>> sji = read_iris_sji_level2_fits(sample.SJI_CUBE_1400)
+    """
+
+    def __init__(self, data, wcs, uncertainty=None, unit=None, meta=None,
```

Comment:
I don't really understand what you mean by "define this method". For now __init__ is just creating the NDCube. 

If I just return the NDCube creation, I must write something else ?

Reply:
as written here, you don't need `__init__` at all. It will just inherit the one from NDCube. You only need to define the `__init__` here if you are putting more code in it than just the `super()` call.

Reply:
Ok so you suggest me that I can remove this part as I don't do anything in the function ! Thanks for your help ! I think I'll let it for now if need but I'll remember it for my later work. Thanks !